### PR TITLE
add JinaRecordEncoder

### PR DESCRIPTION
**Pull request by JoanFM**
## Problem

Depends on `pinecone-text` feature to add JinaEncoder https://github.com/pinecone-io/pinecone-text/pull/59. It aims to add JinaRecordEncoder class to allow users to use Jina Embeddings using Jina Embedding API https://jina.ai/embeddings/

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

### CODE REVIEW

File: src/canopy/knowledge_base/record_encoder/jina.py
Comment (igiloh-pinecone) [inline src/canopy/knowledge_base/record_encoder/jina.py:1]: Hi @JoanFM , please also add a unit tests and a system tests.

Other than that LGTM!

**Review by igiloh-pinecone — CHANGES_REQUESTED** [review]
The only comment as about adding tests

File: src/canopy/knowledge_base/record_encoder/jina.py
Comment (JoanFM) [reply src/canopy/knowledge_base/record_encoder/jina.py:1]: They have been added, please check again

**Review by igiloh-pinecone — APPROVED** [review]
LGTM!

**Pull request merged**